modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp

```diff
@@ -254,6 +273,60 @@ PortletConfigurationPermissionPropagation portletConfigurationPermissionPropagat
 		'<portlet:namespace />saveButton'
 	);
 
+	var <portlet:namespace />permissionPropagationEnabledCheckbox = document.getElementById(
+		'<portlet:namespace />permissionPropagationEnabled'
+	);
+
+	var <portlet:namespace />undoButton = document.getElementById(
+		'<portlet:namespace />undoButton'
+	);
+
+	var alertMessage = document.getElementById('<portlet:namespace />alertMessage');
+
+	if (<portlet:namespace />permissionPropagationEnabledCheckbox) {
+		var initialPermissionPropagationEnabled = Boolean(
+			<%= portletConfigurationPermissionPropagation.getPermissionPropagation(renderRequest) %>
+		);
+
+		<portlet:namespace />permissionPropagationEnabledCheckbox.addEventListener(
+			'click',
+			(event) => {
+				if (!<portlet:namespace />alertMessage) {
+					return;
+				}
+
+				if (
+					(initialPermissionPropagationEnabled || event.target.checked) &&
+					alertMessage.classList.contains('hide')
+				) {
+					alertMessage.classList.remove('hide');
+
+					return;
+				}
+
+				if (
+					!initialPermissionPropagationEnabled &&
+					!event.target.checked &&
+					!alertMessage.classList.contains('hide')
+				) {
+					alertMessage.classList.add('hide');
+				}
+			}
+		);
+	}
+
+	if (<portlet:namespace />undoButton) {
+		<portlet:namespace />undoButton.addEventListener('click', () => {
+			alertMessage.classList.add('hide');
+
+			if (<portlet:namespace />permissionPropagationEnabledCheckbox) {
+				<portlet:namespace />permissionPropagationEnabledCheckbox.checked = Boolean(
+					<%= portletConfigurationPermissionPropagation.getPermissionPropagation(renderRequest) %>
+				);
+			}
+		});
+	}
```

Comment:
I think it could be moved into the if statement `if (<portlet:namespace />permissionPropagationEnabledCheckbox)`